Library/src/test/java/com/shopify/testify/FuzzyCompareTest.kt

```diff
@@ -23,89 +23,133 @@
     }
 
     @Test
-    fun huesIdenticalMod360() {
-        assertFalse(FuzzyCompare.isHueDifferent(0f, 360f, 1.0f))
-        assertFalse(FuzzyCompare.isHueDifferent(360f, 0f, 1.0f))
+    fun differenceLab() {
+        assertEquals(0.2, calculateDeltaE("0F0F0F".lab, "0E0E0E".lab), 0.1)
+        assertEquals(0.4, calculateDeltaE("000000".lab, "000001".lab), 0.1)
+        assertEquals(2.6, calculateDeltaE("ED0000".lab, "E10000".lab), 0.1)
+        assertEquals(33.3, calculateDeltaE("00FF00".lab, "FFFFFF".lab), 0.1)
+        assertEquals(39.7, calculateDeltaE("0000FF".lab, "000000".lab), 0.1)
+        assertEquals(45.8, calculateDeltaE("FF0000".lab, "FFFFFF".lab), 0.1)
+        assertEquals(50.4, calculateDeltaE("FF0000".lab, "000000".lab), 0.1)
+        assertEquals(52.9, calculateDeltaE("FF0000".lab, "0000FF".lab), 0.1)
+        assertEquals(64.2, calculateDeltaE("0000FF".lab, "FFFFFF".lab), 0.1)
+        assertEquals(83.2, calculateDeltaE("00FF00".lab, "0000FF".lab), 0.1)
+        assertEquals(86.6, calculateDeltaE("FF0000".lab, "00FF00".lab), 0.1)
+        assertEquals(87.9, calculateDeltaE("00FF00".lab, "000000".lab), 0.1)
     }
 
     @Test
-    fun huesWithTinyDifferentAnd1PercentTolerance() {
-        for (hue in 0..359) {
-            assertFalse(FuzzyCompare.isHueDifferent(hue.toFloat(), hue + 0.01f, 0.99f))
-        }
+    fun extremeDifferenceLab() {
+        assertEquals(100.0, calculateDeltaE("000000".lab, "FFFFFF".lab), 0.1)
+        assertEquals(100.0, calculateDeltaE("FFFFFF".lab, "000000".lab), 0.1)
+        assertEquals(52.9, calculateDeltaE("FF0000".lab, "0000FF".lab), 0.1)
+        assertEquals(83.2, calculateDeltaE("00FF00".lab, "0000FF".lab), 0.1)
+        assertEquals(86.6, calculateDeltaE("FF0000".lab, "00FF00".lab), 0.1)
     }
 
     @Test
-    fun huesWithTinyDifferentAndNoTolerance() {
-        for (hue in 0..359) {
-            assertTrue(FuzzyCompare.isHueDifferent(hue.toFloat(), hue + 0.01f, 1.0f))
+    fun grayscaleDifferenceLab() {
+        (25..230).forEach { value1 ->
+            val baseColor = RGB(value1, value1, value1)
+            val lab1 = baseColor.toLAB()
+            (-25..25).forEach { value2 ->
+                val currentColor = RGB(value1 + value2, value1 + value2, value1 + value2)
+                val lab2 = currentColor.toLAB()
+                val deltaE = calculateDeltaE(lab1, lab2)
+                if (deltaE >= 10) {
+                    fail("deltaE of #${baseColor.toHex()} and #${currentColor.toHex()} is $deltaE which is greater than 10")
+                }
+            }
         }
     }
 
     @Test
-    fun hues5PercentDifferentWith5PercentTolerance() {
-        for (hue in 0..359) {
-            assertFalse(FuzzyCompare.isHueDifferent(hue.toFloat(), (hue + 18).toFloat(), 0.95f))
+    fun redDifferenceLab() {
+        // Visually very similar Red-based colors
+        (0..20).forEach { value1 ->
+            val baseColor = RGB(255 - value1, value1, value1)
```

Comment:
did you mean to pass `value1` for non Red values? same q for the other colour tests

Reply:
Yes, this is intentional.

This generates visibly "red" colours in the range of `#EB0000` to `#FF1414`. I vary the G and B component by up to 20 because, so long as the R component dominates significantly, that slight amount of other colour still looks very "red".